botorch/optim/optimize.py

```diff
@@ -252,27 +271,64 @@ def _optimize_acqf_sequential_q(
     return candidates, torch.stack(acq_value_list)
 
 
+def _combine_initial_conditions(
+    provided_initial_conditions, generated_initial_conditions, num_restarts
+):
+    if (
+        provided_initial_conditions is not None
+        and generated_initial_conditions is not None
+    ):
+        if (  # Repeat the provided initial conditions to match the number of restarts
+            provided_initial_conditions.shape[0] == 1
+            and num_restarts is not None
+            and num_restarts > 1
+        ):
+            provided_initial_conditions = provided_initial_conditions.repeat(
+                num_restarts, *([1] * (provided_initial_conditions.dim() - 1))
+            )
+        return torch.cat(
+            [provided_initial_conditions, generated_initial_conditions], dim=-2
+        )  # should this be shuffled?
```

Comment:
probably not a bad idea since we're solving some of the optimization in batches by default and it would be good to distribute the provided and generated ones across the different batches.

Reply:
This is now shuffled but there isn't a PRNG seed being set, it looks like the best way to do that would be via `ic_gen_kwargs` but elsewhere in the code `randperm` has been used without seeding and so will leave it to a future refactor/PR if desired